docs/gadget-devel/building.md

```diff
@@ -0,0 +1,106 @@
+
+- `ebpfsource`: eBPF source code file. It defaults to `program.bpf.c`.
+- `metadata`: File containing metadata about the gadget. It defaults to `gadget.yaml`.
+- `wasm`: Wasm module. It is unset by default. This field supports two kind of files:
+    - `*.wasm`: prebuilt Wasm module
+    - `*.go`: automatically built with tinygo
+- `cflags`: The C flags used to compile the eBPF program. It is unset by default.
+
+By default, the build command looks for `build.yaml` in PATH. It can be changed with the `--file` flag:
+
+```bash
+$ ls
+another_name_for_gadget.yaml  another_name_for_program.bpf.c  mybuild.yaml
+
+$ cat mybuild.yaml
+ebpfsource: another_name_for_program.bpf.c
+metadata: another_name_for_gadget.yaml
+
+$ sudo ig image build . -f mybuild.yaml
+INFO[0000] Experimental features enabled
+Successfully built sha256:2f3ccd6254e232e6476f9f015b15f622c44831986f81a82eec17e9c55d98ccaf
+```
+
+## Toolchain location
+
+It is possible to build a gadget using a builder container or by using a local toolchain. By default,
+a container image provided by Inspektor Gadget with all the tools (compiler, libraries,
+header files) needed to compiled your gadgets is used. A different container image can be specified with
+`--builder-image`. This option requires docker to be available on the system.
+
+`--local` can be used to use the tools installed in the local machine. In this case, you'll need to
+have clang, llvm, the gadget headers (see make install-headers) and the [bpf
+headers](https://github.com/libbpf/libbpf/blob/56069cda7897afdd0ae2478825845c7a7308c878/src/Makefile#L160)
+installed.
+
+In this case it's possible to control some of the tools used by setting some env variables:
+
+```bash
+$ sudo CLANG=clang-15 LLVM_STRIP=llvm-strip-15 ig image build . -f mybuild.yaml --local
+```
+
+## Reproducible builds
+
+The `build` command supports the
+[`SOURCE_DATE_EPOCH`](https://reproducible-builds.org/docs/source-date-epoch/)
+env variable:
+
+```bash
+# Set SOURCE_DATE_EPOCH to the last modification of the ebpf program source code.
+# It can be set to any epoch you want.
+$ export SOURCE_DATE_EPOCH="$(date -r program.bpf.c +%s)"
+
+$ sudo -E ig image build -t foo:latest .
+INFO[0000] Experimental features enabled
+Successfully built ghcr.io/inspektor-gadget/gadget/foo:latest@sha256:373f077d366ef2703535e8e862b60f8a35cc1a9312e9e203534b8fce554f8749
+
+# Building again produces the exact same digest
+$ sudo -E ig image build -t foo:latest .
+INFO[0000] Experimental features enabled
+Successfully built ghcr.io/inspektor-gadget/gadget/foo:latest@sha256:373f077d366ef2703535e8e862b60f8a35cc1a9312e9e203534b8fce554f8749
+```
+
+## In-tree gadgets with Wasm
+
+In order to compile the in-tree gadgets (gadgets shipped in the Inspektor gadget
+repository) that use Wasm it's needed to define the `IG_SOURCE_PATH` with the
+absolute path to the the root of the Inspektor Gadget source code.
```

Comment:
Add explanation:
This is to use the latest changes in the wasm api without rebuilding the builder container image.

Reply:
Skipped the container image part as we don't store the wasm API there.